src/win32/findfile.c

```diff
@@ -67,6 +67,48 @@ static wchar_t* win32_walkpath(wchar_t *path, wchar_t *buf, size_t buflen)
 	return (path != base) ? path : NULL;
 }
 
+static int win32_is_git_root(wchar_t *root)
+{
+	/* decide whether this is http://git-for-windows.github.io or http://msysgit.github.io
+	* in git-for-windows the root is either C:\Program Files\Git\mingw64 or C:\Program Files\Git\mingw32, whereas C:\PROGRAMFILES\Git\{cmd,mingw64\bin,mingw32\bin} is on the path
+	* in msysgit the root is C:\Program Files (x86)\Git (on x86 and x64) whereas C:\Program Files (x86)\Git\{bin,cmd} is on the path
```

Comment:
Please, let's get the wording right once and for all, at least in here: msysgit is the development environment to create the MSYS1-based Git for Windows. msysgit is *not* intended for end-users, but for developers who want to work *on* Git for Windows itself.

So I'd propose to change the working as follows:

"git-for-windows" -> "MSYS2-based Git for Windows"
"msysgit" -> "MSYS1-based Git for Windows"

Also, it would be nice to break lines earlier to make the diff more readable on GitHub, and maybe even have proper punctuation (end sentences with a dot, start new sentences with an upper case letter).